### DISCUSSION TITLE
[vue-table] Is there an option to use refs and computed with useVueTable?

### DISCUSSION THREAD

**Discussion by vojvodics** [TanStack/table#5102] [Q&A]
I don't know if this is an issue with the adapter or the types, but it'd be awesome if the useVueTable had a more vue-like API. E.g. `data` should be of type `MaybeRefOrGetter<TData>`, which the adapter can then resolve to `get data() { return toValue(options.data) }`. 

The usage of the table would be more natural in vue:
If you're creating a custom component:
```
useVueTable({
  data: () => props.data, // vue 3.3, or if using lower version computed(() => props.data)
  state: () => ({ ...your state }),
});
```
If using vue-query
```
const { data } = useQuery();
const table = useVueTable({ data });
```

*There is a related unanswered discussion (https://github.com/TanStack/table/discussions/4455) which probably highlights the confusion around the current vue-table API

**Comment by VincEnterprise** [comment]
Bump I would like to have this feature as well

**Comment by OlaAlsaker** [comment]
I have now created a PR which will add support for reactive data in the Vue-adapter: https://github.com/TanStack/table/pull/5687
